src/idpi/operators/vertical_reduction.py

```diff
@@ -317,8 +317,8 @@ def integrate_k(field, operator, mode, height, h_bounds, hsurf=None):
     if "generalVerticalLayer" in field_in_h_bounds.dims:
         rfield = (
             (field_in_h_bounds * dh_in_h_bounds)
-            .fillna(0.0)
             .sum(dim="generalVerticalLayer")
+            .where(~field_in_h_bounds.isnull().all(dim="generalVerticalLayer"))
```

Comment:
is the order here correct? Isnt the idea to eliminate the nan before sum ?

Reply:
The idea is to keep full columns of nans as nans in the output field. It seems that fieldextra is doing it this way but I haven't confirmed this.

Reply:
Yes, fieldextra keeps full columns of nans as nans in the output field. It is left to the user to reset nan values by specific values, which he can do by specifying poper="replace_undef,<value>" in the namelist (<value> to be replaced by a real number).

Reply:
I added a comment